### PR TITLE
Update stats inclusion to use proxy config instead

### PR DESCRIPTION
**Pull request by bianpengyuan**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 1]
```diff
@@ -21,30 +21,57 @@ To see the statistics for a pod:
 $ kubectl exec $POD -c istio-proxy -- pilot-agent request GET stats
 {{< /text >}}
 
-See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats)
-for an explanation of the data recorded.
+Envoy collects various kinds of statistics. The following links are Envoy documents about some stats that it collects:
```
Comment (douglas-reid) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:24]:
```suggestion
Envoy generates statistics about its behavior, scoping the statistics by function, including:
```

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 2]
```diff
@@ -21,30 +21,57 @@ To see the statistics for a pod:
...
 
-By default, Istio configures Envoy to record minimal statistics.  The default collection
+- [Upstream connection](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats)
+- [Listener](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/stats)
+- [HTTP Connection Manager](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/stats)
+- [TCP proxy](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/tcp_proxy_filter#statistics)
+- [Router](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter.html?highlight=vhost#statistics)
+
+By default, Istio configures Envoy to record minimal statistics. The default collection
```
Comment (douglas-reid) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:32]:
```suggestion
By default, Istio configures Envoy to record a minimal set of statistics to reduce the overall CPU and memory footprint of installed proxies. The default collection
```

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 3]
```diff
@@ -21,30 +21,57 @@ To see the statistics for a pod:
...
 the `stats_matcher` JSON element.
 
-To Configure Envoy to record statistics for inbound or outbound traffic, add the
-`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in the Kubernetes `Deployment`.
-Add the `cluster.outbound` prefix to gather data about outbound traffic activity and circuit breaking.
-To gather data on inbound traffic, add the `listener` prefix.  The sample
-[fortio-deploy.yaml]({{< github_file>}}/samples/httpbin/sample-client/fortio-deploy.yaml)
-shows use of `sidecar.istio.io/statsInclusionPrefixes` with the `cluster.outbound` prefix.
+{{< tip >}}
+Note Envoy stats name highly depends on how Envoy configuration is composed, and thus could tie to Istio control plane implementation detail.
```
Comment (douglas-reid) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:48]:
```suggestion
Note: The names of Envoy statistics can vary based on the composition of Envoy configuration. As a result, the exposed names of statistics for Envoys managed by Istio are subject to the configuration behavior of Istio.
```

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 4]
```diff
@@ -21,30 +21,57 @@ To see the statistics for a pod:
...
+If you are building dashboard or alert based on Envoy stats, before upgrading your Istio, it is highly recommended to examine the stats in a canary environment.
```
Comment (douglas-reid) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:49]:
```suggestion
If you build or maintain dashboards or alerts based on Envoy statistics, it is **strongly recommended** to examine the statistics in a canary environment **before upgrading Istio**. 
```

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 5]
```diff
@@ -21,30 +21,57 @@ To see the statistics for a pod:
...
+{{< /tip >}}
+
+To configure Istio proxy to record additional statistics, you can add [`ProxyConfig.ProxyStatsMatcher`](/docs/reference/config/istio.mesh.v1alpha1/#ProxyStatsMatcher) to your mesh config. For example, to enable stats for circuit breaker, retry, and upstream connections globally, you can specify stats matcher as follow:
```
Comment (douglas-reid) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:52]: does this require pod restarts, etc?  What can the user expect if they manage to update?

Comment (bianpengyuan) [reply content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:53]: Added a note about this. Thanks!

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 6]
```diff
@@ -21,30 +21,62 @@ To see the statistics for a pod:
...
 
-To Configure Envoy to record statistics for inbound or outbound traffic, add the
-`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in the Kubernetes `Deployment`.
-Add the `cluster.outbound` prefix to gather data about outbound traffic activity and circuit breaking.
-To gather data on inbound traffic, add the `listener` prefix.  The sample
-[fortio-deploy.yaml]({{< github_file>}}/samples/httpbin/sample-client/fortio-deploy.yaml)
-shows use of `sidecar.istio.io/statsInclusionPrefixes` with the `cluster.outbound` prefix.
+{{< tip >}}
+Note: The names of Envoy statistics can vary based on the composition of Envoy configuration. As a result, the exposed names of statistics for Envoys managed by Istio are subject to the configuration behavior of Istio.
+If you build or maintain dashboards or alerts based on Envoy statistics, it is **strongly recommended** to examine the
```
Comment (Rachael-Graham) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:49]:
```suggestion
If you build or maintain dashboards or alerts based on Envoy statistics, it is **strongly recommended** that you examine the
```

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 7]
```diff
@@ -21,30 +21,62 @@ To see the statistics for a pod:
...
+statistics in a canary environment **before upgrading Istio**.
+{{< /tip >}}
+
+To configure Istio proxy to record additional statistics, you can add [`ProxyConfig.ProxyStatsMatcher`](/docs/reference/config/istio.mesh.v1alpha1/#ProxyStatsMatcher) to your mesh config. For example, to enable stats for circuit breaker, retry, and upstream connections globally, you can specify stats matcher as follow:
+
+{{< tip >}}
+Proxy needs to restart to pick up the stats matcher configuration.
```
Comment (Rachael-Graham) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:56]: do users need to manually restart? if yes i would specify

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 8]
```diff
@@ -21,30 +21,62 @@ To see the statistics for a pod:
...
+{{< text yaml >}}
+proxyStatsMatcher:
+  inclusionRegexps:
+    - ".*circuit_breakers.*"
+  inclusionPrefixes:
+    - "upstream_rq_retry"
+    - "upstream_cx"
+{{< /text >}}
+
+You can also override the global stats matching configuration at per proxy by using `proxy.istio.io/config` annotation. For example, to configure the same stats generation inclusion as above, you can add the annotation to a gateway proxy or a workload as follow:
```
Comment (Rachael-Graham) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:68]:
```suggestion
You can also override the global stats matching configuration per proxy by using the `proxy.istio.io/config` annotation. For example, to configure the same stats generation inclusion as above, you can add the annotation to a gateway proxy or a workload as follow:
```

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md [hunk 9]
```diff
@@ -21,30 +21,62 @@ To see the statistics for a pod:
...
+    inclusionPrefixes:
+    - "upstream_rq_retry"
+    - "upstream_cx"
+{{< /text >}}
 
-You can override the Envoy defaults to gather less data than usual.  Use
-`sidecar.istio.io/statsInclusionPrefixes: cluster_manager,listener_manager`
-to collect the least statistics possible.
+{{< tip >}}
+Note if you are using `sidecar.istio.io/statsInclusionPrefixes`, `sidecar.istio.io/statsInclusionRegexps`, and `sidecar.istio.io/statsInclusionSuffixes`, please consider switching to `ProxyConfig` based configuration as it provides global default and uniform way to override at gateway and sidecar proxy.
```
Comment (Rachael-Graham) [inline content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:81]:
```suggestion
Note: If you are using `sidecar.istio.io/statsInclusionPrefixes`, `sidecar.istio.io/statsInclusionRegexps`, and `sidecar.istio.io/statsInclusionSuffixes`, consider switching to the `ProxyConfig`-based configuration as it provides a global default and a uniform way to override at the gateway and sidecar proxy.
```

**Review by Rachael-Graham — COMMENTED** [review]
docs review for style/clarity/etc. minor suggestions
cc @frankbu

File: content/en/docs/ops/configuration/telemetry/envoy-stats/index.md (hunk 7 above)
Comment (bianpengyuan) [reply content/en/docs/ops/configuration/telemetry/envoy-stats/index.md:56]: I am not sure what is categorized as `manual restarted`. Any restart of pod will pick up the configuration. This is just to set up expectation to the user that this won't take effect at runtime.

**Review by frankbu — APPROVED** [review]

**Pull request merged**